packages/react-native/src/primitives/Checkbox/Checkbox.tsx

```diff
@@ -67,22 +57,23 @@ export default function Checkbox<T>({
     [buttonStyle]
   );
 
+  // TODO: replace IconButton with icon once Icon primitive is added
+
   return (
-    <View style={[containerStyle, style]}>
-      {label && labelPrecedesIcon ? (
-        <Label style={labelStyle}>{label}</Label>
-      ) : null}
+    <Pressable
+      accessibilityRole={accessibilityRole}
+      disabled={disabled}
+      onPress={handleOnChange}
+      style={[containerStyle, style]}
+    >
       <IconButton
         {...rest}
-        disabled={disabled}
-        onPress={handleOnChange}
-        source={selected ? icons.checkboxFilled : icons.checkboxOutline}
+        disabled
+        source={checked ? icons.checkboxFilled : icons.checkboxOutline}
         size={size}
         style={iconButtonStyle}
```

Comment:
Let's make sure we clean up the `buttonStyle` prop and `iconButtonStyle` when we replace the `IconButton`